covscript/src/tokens.ts

```diff
@@ -0,0 +1,204 @@
+  name: "StringLiteral",
+  // A string literal, taking care of \" inside the string.
+  pattern: /"(:?[^\\"]|\\(:?[bfnrtv"\\/]|u[0-9a-fA-F]{4}))*"/,
+});
+
+export const NumberLiteral = createToken({
+  name: "NumberLiteral",
+  // All kinds of number, integer or decimal, negative or positive, exponential notation supported.
+  pattern: /-?(0|[1-9]\d*)(\.\d+)?([eE][+-]?\d+)?/,
+});
+
+export const Identifier = createToken({
+  name: "Identifier",
+  // Alphanumeric, can not start with number, length at least 1.
+  // Can contain dots, per convention
+  pattern: /[a-zA-Z_][\w\.]*/,
+});
+
+export const True = createToken({ name: "True", pattern: /true|True|TRUE/ });
+export const False = createToken({
+  name: "False",
+  pattern: /false|False|FALSE/,
+});
+export const Null = createToken({ name: "Null", pattern: /null|Null|NULL/ });
+
+export const LSquare = createToken({ name: "LSquare", pattern: /\[/ });
+export const RSquare = createToken({ name: "RSquare", pattern: /\]/ });
+
+export const LRound = createToken({ name: "LRound", pattern: /\(/ });
+export const RRound = createToken({ name: "RRound", pattern: /\)/ });
+
+export const If = createToken({
+  name: "If",
+  pattern: /if|IF|If/,
+  categories: Keyword,
+});
+export const Then = createToken({
+  name: "Then",
+  pattern: /then|THEN|Then/,
+  categories: Keyword,
+});
+export const Else = createToken({
+  name: "Else",
+  pattern: /else|ELSE|Else/,
+  categories: Keyword,
+});
+export const EndIf = createToken({
+  name: "EndIf",
+  pattern: /endif|ENDIF|EndIf/,
+  categories: Keyword,
+});
+
+export const Comma = createToken({ name: "Comma", pattern: /,/ });
+export const Add = createToken({
+  name: "Add",
+  pattern: /\+/,
+  categories: AdditionOperator,
+});
+export const Sub = createToken({
+  name: "Sub",
+  pattern: /-/,
+  categories: AdditionOperator,
+});
+
+export const Mult = createToken({
+  name: "Mult",
+  pattern: /\*/,
+  categories: MultiplicationOperator,
+});
+export const Div = createToken({
+  name: "Div",
+  pattern: /\/|\÷/,
+  categories: MultiplicationOperator,
+});
+export const Mod = createToken({
+  name: "Mod",
+  pattern: /%/,
+  categories: MultiplicationOperator,
+});
+
+export const InOperator = createToken({
+  name: "In",
+  pattern: /IN|in|In/,
+  categories: ComparisonOperator,
+});
+
+export const Greater = createToken({
+  name: "Greater",
+  pattern: />/,
+  categories: ComparisonOperator,
+});
+export const Less = createToken({
+  name: "Less",
+  pattern: /</,
+  categories: ComparisonOperator,
+});
+export const GreaterEqual = createToken({
+  name: "GreaterEqual",
+  pattern: />=/,
+  categories: ComparisonOperator,
+});
+export const LessEqual = createToken({
+  name: "LessEqual",
+  pattern: /<=/,
+  categories: ComparisonOperator,
+});
+export const Equal = createToken({
+  name: "GreaterEqual",
+  pattern: /==/,
+  categories: ComparisonOperator,
+});
+export const NotEqual = createToken({
+  name: "LessEqual",
```

Comment:
Name should probably be `NotEqual`?
```suggestion
  name: "NotEqual",
```

Reply:
Done.